pkg/remote/aws/sqs_queue_policy_supplier.go

```diff
@@ -0,0 +1,65 @@
+package aws
+
+import (
+	"github.com/cloudskiff/driftctl/pkg/remote/aws/repository"
+	"github.com/cloudskiff/driftctl/pkg/remote/deserializer"
+	remoteerror "github.com/cloudskiff/driftctl/pkg/remote/error"
+	"github.com/cloudskiff/driftctl/pkg/resource"
+	"github.com/cloudskiff/driftctl/pkg/resource/aws"
+	awsdeserializer "github.com/cloudskiff/driftctl/pkg/resource/aws/deserializer"
+	"github.com/cloudskiff/driftctl/pkg/terraform"
+	"github.com/sirupsen/logrus"
+	"github.com/zclconf/go-cty/cty"
+)
+
+type SqsQueuePolicySupplier struct {
+	reader       terraform.ResourceReader
+	deserializer deserializer.CTYDeserializer
+	client       repository.SQSRepository
+	runner       *terraform.ParallelResourceReader
+}
+
+func NewSqsQueuePolicySupplier(provider *TerraformProvider) *SqsQueuePolicySupplier {
+	return &SqsQueuePolicySupplier{
+		provider,
+		awsdeserializer.NewSqsQueuePolicyDeserializer(),
+		repository.NewSQSClient(provider.session),
+		terraform.NewParallelResourceReader(provider.Runner().SubRunner()),
+	}
+}
+
+func (s SqsQueuePolicySupplier) Resources() ([]resource.Resource, error) {
+	queues, err := s.client.ListAllQueues()
+	if err != nil {
+		return nil, remoteerror.NewResourceEnumerationErrorWithType(err, aws.AwsSqsQueuePolicyResourceType, aws.AwsSqsQueueResourceType)
```

Comment:
I think you should swap types arguments like this : ` remoteerror.NewResourceEnumerationErrorWithType(err, aws.AwsSqsQueueResourceType, aws.AwsSqsQueuePolicyResourceType)`

Reply:
From how @moadibfr did it on the S3BucketAnalytics it seems to be the supplier first where the error occurred and then the type of the listed resources.

```go
func NewResourceEnumerationErrorWithType(error error, supplierType string, listedTypeError string) *ResourceEnumerationError {
	context := map[string]string{
		"ListedTypeError": listedTypeError,
	}
	return &ResourceEnumerationError{
		SupplierError:   *NewSupplierError(error, context, supplierType),
		listedTypeError: listedTypeError,
	}
}
```

Reply:
You're right my bad !